### PR TITLE
Update

### PR DESCRIPTION
**Pull request by Jake-Shadle**
Updates all dependencies, and reverts #80 until `procfs-core` is released. Requested in https://github.com/eminence/procfs/issues/273.

Also fixed the `test_merged_mappings` which was failing on my machine due to the ` (deleted)` suffix causing the path comparison to fail. It now just checks the beginning matches which should effectively be the same unless this crate's testing gets far more comprehensive in the future.

### CODE REVIEW

File: Cargo.toml
Comment (gabrielesvelto) [inline Cargo.toml:37]: I'd rather have us revert https://github.com/rust-minidump/minidump-writer/commit/b95bf2e31f0be429a78f9639835312d8423fd632 instead of falling back on vanilla `procfs` because we can't vendor it in Firefox due to its broad dependencies (including windows-sys which we can't still vendor). If that's not a problem I'll re-land the switch to `procfs-core` as soon as they cut a proper release.

File: Cargo.toml
Comment (gabrielesvelto) [inline Cargo.toml:25]: rust-minidump seems to be pulling in 0.5.7 and that's part of the reason for the failures on macOS, I'm going to update it

File: Cargo.toml
Comment (gabrielesvelto) [inline Cargo.toml:56]: We need to add `minidump-unwind` as a dependency as some of the functionality we're using in the tests has been split out into that crate.

**Review by gabrielesvelto — CHANGES_REQUESTED** [review]

File: Cargo.toml
Comment (Jake-Shadle) [reply Cargo.toml:37]: Ahh ok I'll wait for a response from the maintainers (probably US? timezone) today and if they can't make a release can revert that.

File: Cargo.toml
Comment (gabrielesvelto) [reply Cargo.toml:25]: Or we do the opposite and go back to 0.5.x until they update first.

Comment (Jake-Shadle) [reply Cargo.toml]: Yah that's what I opted for, easiest.

File: Cargo.toml
Comment (gabrielesvelto) [inline Cargo.toml:15]: You just need to bump crash-context to `0.6.1` and we're done, thank you!

**Review by gabrielesvelto — CHANGES_REQUESTED** [review]

File: Cargo.toml
Comment (Jake-Shadle) [reply Cargo.toml:15]: 0.6.1 is semver compatible with 0.6, I tend to leave off compatible versions since it just leads to churn in the manifest file, but I'll add it for clarity.

**Review by gabrielesvelto — APPROVED** [review]

**Comment by eminence** [comment]
Hi all, just dropping a quick note to say that `procfs-core` v0.16.0 has now been published.  You are currently depending on v0.16.0-RC1.   The difference between the two is very small, just documentation fixes, and two new feature (parsing `/proc/net/snmp` data and `/proc/partitions` data) that won't impact you

**Pull request merged**
